### DISCUSSION TITLE
Progress modal

### DISCUSSION THREAD

**Discussion by tmarsik42** [carbon-design-system/carbon-components-svelte#1952] [Ideas]
I've been going through the Carbon Design System documentation and came across the [Progress Modal](https://carbondesignsystem.com/components/modal/usage/#progress-modal) section. The design implementation there seems quite useful (though I'm not sure if it's specific to Carbon v11).

Personally, I find progress modals quite handy and having a straightforward API to switch between modal types would be a great addition. Currently, we have danger and passive modal props. I've been using the option described [here](https://carbon-components-svelte.onrender.com/components/Modal#multiple-secondary-buttons), but it does have its drawbacks. For instance, having two secondary buttons for similar actions (both essentially dismiss) might confuse users.

**Comment by metonym** [comment]
Example: https://svelte.carbondesignsystem.com/components/ComposedModal#progress-modal

<img width="1085" height="591" alt="Screenshot 2026-02-16 at 12 03 31 PM" src="https://github.com/user-attachments/assets/0dcd8de1-823e-4bb6-ab58-aab36ad2309f" />
